packages/router/router.ts

```diff
@@ -3369,6 +3365,21 @@ export function getStaticContextFromError(
   return newContext;
 }
 
+function throwStaticHandlerAbortedError(
+  request: Request,
+  isRouteRequest: boolean,
+  future: StaticHandlerFutureConfig
+) {
+  let method = isRouteRequest ? "queryRoute" : "query";
+  if (future.v7_throwAbortReason && request.signal.reason !== undefined) {
+    throw request.signal.reason;
```

Comment:
The only downside I see here is that we lose the request method/URL that we include in the custom error, but I think if folks need that then they should be passing a custom reason when they call `controller.abort`.  Right now, in Remix we don't expose that for customization - but we could send our own `DOMException` with the method/URL in the message if we wanted to.